File: src/main/java/dev/mantasboro/trakt5/entities/Metadata.java
package dev.mantasboro.trakt5.entities;

import com.google.gson.annotations.SerializedName;
import dev.mantasboro.trakt5.enums.*;

public class Metadata {

    public MediaType media_type;
    public Resolution resolution;
    public Hdr hdr;
    public Audio audio;
    public AudioChannels audio_channels;
    @SerializedName("3d")
    public Boolean is3d;

}

File: src/main/kotlin/dev/mantasboro/trakt5/entities/Metadata.kt
package dev.mantasboro.trakt5.entities

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import dev.mantasboro.trakt5.enums.Audio
import dev.mantasboro.trakt5.enums.AudioChannels
import dev.mantasboro.trakt5.enums.Hdr
import dev.mantasboro.trakt5.enums.MediaType
import dev.mantasboro.trakt5.enums.Resolution

@JsonClass(generateAdapter = true)
data class Metadata(
    val media_type: MediaType?,
    val resolution: Resolution?,
    val hdr: Hdr? = null,
    val audio: Audio? = null,
    val audio_channels: AudioChannels? = null,

    @Json(name = "3d")
    val is3d: Boolean? = null,
)
